xrpl/models/base_model.py

```diff
@@ -54,7 +72,18 @@ def _key_to_json(field: str) -> str:
         1. 'TransactionType' becomes 'transaction_type'
         2. 'value' remains 'value'
         3. 'URI' becomes 'uri'
+
+        This function accepts inputs in camelCase only
+
+    Raises:
+        XRPLModelException: If the input is invalid
     """
+    if is_snake_case(field):
```

Comment:
I think this should actually be checking for only `PascalCase` - we don't want `kebab-case`, `camelCase`, `flatcase`, or anything else either.

Reply:
Okay. There are certain fields like `currency`, `issuer` and `value` in the `TakerPays` (`OfferCreate` Transaction) --> These fields do not adhere to PascalCase.

I'm using this regex to match for PascalCase: `re.compile("^[A-Z][a-z]+(?:[A-Z][a-z]+)*$")`

I can move this check inside `BaseModel.from_xrpl` (not `_key_to_json`). This would not enforce the PascalCase check on every dictionary

Reply:
That's fair, we can account for those as well (`camelCase` and `pascalCase`)

Reply:
fixed in [7405ba3](https://github.com/XRPLF/xrpl-py/pull/694/commits/7405ba3e197f840519629679d2da418639d5e0d6)